gobblin-salesforce/src/main/java/org/apache/gobblin/salesforce/SalesforceExtractor.java

```diff
@@ -896,6 +928,43 @@ private void reinitializeBufferedReader() throws IOException, AsyncApiException
     }
   }
 
+  private List<JsonElement> fetchWithRetry(
```

Comment:
bulk api may have retry internally?

Reply:
Bulk API has retry. It means if the result set is too big, it would retry 15 times creating/caching the result set
This retry is our logic. When we fetch the result set file, if there is any failure, we want to retry specified times. It won't hurt. if the result file can be fetch at first time, the function returns the result set.

Reply:
If bulk api already does retry, can we remove the retry on our side? Application layer retry after api layer retries mostly won't change the error result if any.

Reply:
sfdc retrying is `to prepare and create result set files from query data`. 
After sfdc, create the result set files, we request to fetch/download the files.
Our retrying is for fetch/download files.

Reply:
Closing the thread. Please reopen it, if necessary.